### PR TITLE
Extend workshop with a section about `jq` filters

### PR DESCRIPTION
**Pull request by yuumasato**
- Add continuity links for the sections.
- Add section for `jq` filters
- Edit occurences of `add_platform_rule.py` to `add_kubernetes_rule.py`, following https://github.com/ComplianceAsCode/content/pull/11194

### CODE REVIEW

File: doc/tutorials/workshop/content/exercises/13-complex-yaml.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+$ ./utils/add_kubernetes_rule.py create platform \
+    --rule check_nested_yaml \
+    --name my-nested-compliance-configmap --namespace openshift --type configmap \
+    --title "Check value of nested-key in my-config-yaml" \
+    --description "It is important that the nested configmap has a nested-key with value nested-compliant." \
+    --match-entity "at least one" \
+    --match "nested-compliant" \
+    --yamlpath ".nested-key" \
+    --jqfilter '.data."my-config.yaml"'
+```
```
Comment (yuumasato) [inline doc/tutorials/workshop/content/exercises/13-complex-yaml.md:84]: I think I need to explain what the command did and what is on the `rule.yml`

File: doc/tutorials/workshop/content/exercises/13-complex-yaml.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+Title: Checking complex yaml structures
+PrevPage: 12-common-rules.md
+NextPage: ../finish
+---
+
+Checking complex yaml structures
+====================
+
+The `yamlfile_value` template is great to check if a yaml key exists or not,
+and whether it exists has a specific value.
```
Comment (rhmdnd) [inline doc/tutorials/workshop/content/exercises/13-complex-yaml.md:11]: nit: and if it does exist, if it has a specific value.

File: doc/tutorials/workshop/content/exercises/13-complex-yaml.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+But some of the resources and configurations of a Kubernetes cluster can be
+defined in quite a complex way, and just the Yaml path syntax used by the
```
Comment (rhmdnd) [inline doc/tutorials/workshop/content/exercises/13-complex-yaml.md:14]: nit: s/Yaml/yaml/ to be consistent with other usage in the file.

File: doc/tutorials/workshop/content/exercises/13-complex-yaml.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+template may not be sufficient to get to the value we want to assess.
+
+For this reason the CaC/content rules used by the Compliance Operator can
+leverage [`jq`](https://jqlang.github.io/jq/) to better select what needs to be assessed.
+A `jq` filter allows us to pre-process the resources and configurations so that
+they are easier to check with the `yamlfile_value` template.
```
Comment (rhmdnd) [inline doc/tutorials/workshop/content/exercises/13-complex-yaml.md:20]: 💯 

Thanks for adding this.

File: doc/tutorials/workshop/content/exercises/13-complex-yaml.md (hunk 1 above)
Comment (rhmdnd) [reply doc/tutorials/workshop/content/exercises/13-complex-yaml.md:84]: A diagram might be worth a thousand words here. The jq filter simplifies what gets written into a file that that scanner is evaluating.

File: doc/tutorials/workshop/content/exercises/13-complex-yaml.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+```
+$ ./utils/add_platform_rule.py cluster-test --rule check_nested_yaml
+```
+
+## 2. Filtering the data to have simpler yamlpaths
+
+This is a generalization of the first use case, `jq` filters can be used to filter and select the data
+fetched by the operator.
+
+When the resource being checked is extensive or complex, `jq` is prety powerful and using it to filter and trim down
```
Comment (rhmdnd) [inline doc/tutorials/workshop/content/exercises/13-complex-yaml.md:96]: Technically, we wouldn't  have any other way to do extensive or complex checking without jq.

If so, we could make this sentence more assertive:

When the resource being checked is an extensive or complex, `jq` is invaluable for simplifying the data before it's evaluated with `yamlpath`.

**Review by rhmdnd — COMMENTED** [review]
This is great - thank you for adding it.

I only have minor comments inline that could be addressed in a follow up.

File: doc/tutorials/workshop/content/exercises/13-complex-yaml.md (hunk 1 above)
Comment (yuumasato) [reply doc/tutorials/workshop/content/exercises/13-complex-yaml.md:95]: I added a diagram, please check it out.
It maybe it easier to check it on the rendered file:
https://github.com/ComplianceAsCode/compliance-operator/blob/4a1d2740d61b5a08b3ce8fae0787002d32cab55d/doc/tutorials/workshop/content/exercises/13-complex-yaml.md

**Comment by yuumasato** [comment]
I have reacted to feedback and improved some of the section:
https://github.com/ComplianceAsCode/compliance-operator/blob/4a1d2740d61b5a08b3ce8fae0787002d32cab55d/doc/tutorials/workshop/content/exercises/13-complex-yaml.md

**Review by rhmdnd — APPROVED** [review]
/lgtm

**Pull request merged**